src/transaction/log_applier_sql_log.c

```diff
@@ -626,26 +620,38 @@ sl_open_next_file (FILE * old_fp)
 }
 
 /*
- * sl_remove_oldest_file() - remove oldest sql log file
- *   return: error code
+ * sl_remove_oldest_file_if_needed() - remove oldest sql log file
  *
  * Note:
  *   This function is related to the ha_sql_log_max_count system parameter.
  *   This system parameter can be set from 2 to 5 and only that number of sql log files are kept.
  */
-static int
-sl_remove_oldest_file (void)
+static void
+sl_remove_oldest_file_if_needed (void)
 {
   unsigned int oldest_file_id;
   char oldest_file_path[PATH_MAX];
 
-  oldest_file_id = sl_Info.curr_file_id - sql_log_max_cnt;
+  // step(1) : guess oldest file
+  // If the file ID is smaller than 'sql_log_max_cnt', it is considered to have wrapped around after exceeding 'UINT_MAX'.
+  if (sl_Info.curr_file_id < sql_log_max_cnt)
+    {
+      oldest_file_id = UINT_MAX + sl_Info.curr_file_id - sql_log_max_cnt + 1;
```

Comment:
NIT:

UINT_MAX + sl_Info.curr_file_id 계산에 의해 중간 결과가 overflow 발생하지 않도록 다음과 같이 수정

```suggestion
      oldest_file_id = UINT_MAX - sql_log_max_cnt + sl_Info.curr_file_id + 1;
```

Reply:
수정하였습니다.